src/main/java/org/jboss/xavier/sanitizer/XavierPayloadSanitizer.java

```diff
@@ -0,0 +1,169 @@
+    {
+        if (inputFile == null)
+        {
+            throw new RuntimeException(description + "path not specified");
+        }
+        try
+        {
+            return new String(Files.readAllBytes(Paths.get(inputFile)), StandardCharsets.UTF_8);
+        }
+        catch(IOException ioe)
+        {
+            throw new RuntimeException(description + "could not be read");
+        }
+    }
+
+
+
+    private static List<String> splitConditions(String inputConditionsString, String sectionHeader)
+    {
+        if (inputConditionsString == null)
+        {
+            throw new RuntimeException("No issues conditions found");
+        }
+        List<String> returnIssues = new ArrayList<>();
+        String[] issues = inputConditionsString.split("\\R");
+        boolean currentSectionWanted = false;
+        for (String issue:issues)
+        {
+            if(issue.equals(sectionHeader))
+            {
+                currentSectionWanted = true;
+                continue;
+            }else if (issue.startsWith("###"))
+            {
+                currentSectionWanted = false;
+                continue;
+            }
+            if(currentSectionWanted)
+            {
+                returnIssues.add(issue);
+            }
+        }
+
+        return returnIssues;
+
+    }
+
+    private static String sanitize(String fileToBeSanitized, List<String> vmsIssuesConditions, List<String> hostsIssuesConditions, List<String> clustersIssuesConditions)
+    {
+        String vmsInvalidElementIds = findInvalidElements(fileToBeSanitized, vmsIssuesConditions);
+        String hostsInvalidElementIds = findInvalidElements(fileToBeSanitized, hostsIssuesConditions);
+        String clustersInvalidElementIds = findInvalidElements(fileToBeSanitized, clustersIssuesConditions);
+
+        String resultString = retireVms(fileToBeSanitized,"$.ManageIQ::Providers::Vmware::InfraManager[*].vms[?(@.id in [" + vmsInvalidElementIds+ "])]");
+        resultString = retireVms(resultString,"$.ManageIQ::Providers::Vmware::InfraManager[*].vms[?(@.host.ems_ref in [" + hostsInvalidElementIds+ "])]");
+
+
+
+        return resultString;
+    }
+
+
+
+    private static String findInvalidElements(String fileToBeSanitized, List<String> issuesConditions)
+    {
+        Configuration conf = Configuration.builder().options(Option.ALWAYS_RETURN_LIST, Option.SUPPRESS_EXCEPTIONS).build();
+        DocumentContext parsedFile = JsonPath.using(conf).parse(fileToBeSanitized);
+
+        List<Object> errorIds = new LinkedList<>();
+
+        issuesConditions.forEach(issuesCondition -> {
+            errorIds.addAll(parsedFile.read(issuesCondition));
+        });
+
+        List<String> errorIdStringsList = new ArrayList<>();
+
+        errorIds.forEach(id -> errorIdStringsList.add(String.valueOf(id)));
+
+        return String.join(",",errorIdStringsList);
+    }
+
+    private static String retireVms(String fileToBeSanitized, String vmSelectionQuery)
```

Comment:
It seems that this method is not going to retire or remove a node but just add a field to mark a node as retired. Could we enhance the name of this method, please? something like `markVmsNodesAsRequired` or any other meaningful name you can think of.

Reply:
method renamed